src/_pytest/monkeypatch.py

```diff
@@ -194,7 +196,11 @@ def delattr(self, target, name=notset, raising=True):
             if raising:
                 raise AttributeError(name)
         else:
-            self._setattr.append((target, name, getattr(target, name, notset)))
+            oldval = getattr(target, name, notset)
```

Comment:
this inadvertently triggers misbehaviours

a) ignores the MRO
b) always goes the normal path, even for classes

Reply:
I'm confused how this is different from the current behavior of setattr. I believe this works the exact same as before, except in the case of the class, where we match the behavior of setattr.

Is there an example you can provide to correct my misunderstanding?

Reply:
I think I understand what you be regarding b. I mimicked the implementation for [`monkeypatch.setattr`](https://github.com/pytest-dev/pytest/blob/master/src/_pytest/monkeypatch.py#L166-L168). I can update both of them if you want to use an if-else.

Reply:
@RonnyPfannschmidt any more comments here?